### PR TITLE
Progress on clean and feature engineering modules

### PR DESCRIPTION
**Pull request by morganmayer**

### CODE REVIEW

**Review by kyleniemeyer — COMMENTED** [review]
You may want to add a Python `.gitignore` file, so that the `.pyc` files and `__pycache__` folders don't get caught in your commits; you'll also want to manually remove those files using `git rm`

File: test.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import caml
+
+def test_read_input():
+
+    test_input = "input.yaml"
+    input = read_input(test_input)
+
```
Comment (lilleybe) [inline test.py]: should there be an assertion here?

File: caml.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import argparse
+import yaml
+
+# caml modules
+import clean
+
+""" This module is meant for control sequence """
+
+def read_input(input_file):
+
```
Comment (lilleybe) [inline caml.py]: doc string goes below def?

Comment (kyleniemeyer) [reply caml.py]: you can have a docstring for a module, but it should be the very first line, I think

File: clean.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import matplotlib.pyplot as plt
+
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.preprocessing import MinMaxScaler
+
+""" This module is meant for data cleaning """
+
+def normalize_overall_min_max(df_train, df_test, target):
+    ''' This function takes the train and test dataframes and
```
Comment (lilleybe) [inline clean.py:12]: Doc strings can be expanded after the module is complete, but I think the format is a single line brief description, then a empty line, then a longer description, then a description of inputs, and then outputs.

File: feature_engineering.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    principalDf_train = pd.DataFrame(data = principalComponents_train
+             , columns = col_list, index = index_list_train)
+    
+    principalComponents_test = pca.transform(x_test)
+    
+    principalDf_test = pd.DataFrame(data = principalComponents_test
+             , columns = col_list, index = index_list_test)
+
+    df_train_transformed = pd.merge(df_train[[str(target)]], principalDf_train, how='outer', left_index=True, right_index=True)
```
Comment (lilleybe) [inline feature_engineering.py:37]: might want to turn these into two-liners (too many columns)

**Review by lilleybe — APPROVED** [review]
Two minor comments on docstrings and too many columns. Looks good!

**Pull request merged**